### PR TITLE
Don't recreate manifests if they exist

### LINKED ISSUE #1679: Only create manifest files if they do not already exist
FEATURE REQUEST

Currently the **certs** and **kubeconfig** phases will skip creation of files if they already exist, preserving user changes; however, **control-plane** and **etcd** phases always recreate the manifest files. 

This creates problems for my workflow because I'm currently manually patching the generated manifests in order to fix the securityContext for selinux. I can skip the preflight check for these manifest files, but my files get replaced. I work around this currently with **--skip-phases**.

I feel this is unexpected behavior as it is inconsistent with the other phases. I propose that the **control-plane** and **etcd** phases have the same behavior as **certs** and **kubeconfig** phases with respect to treatment of existing files.

## Versions

**kubeadm version** (use `kubeadm version`): 

## What happened?
When running `kubeadm init phase control-plane` or `kubeadm init phase etcd local` the manifest files are always regenerated.

## What you expected to happen?
A log message would be printing whining about the files existence, but the existing files would not be replaced.

## How to reproduce it (as minimally and precisely as possible)?
1. On a new machine
2. Place a valid manifest file in /etc/kubernetes/manifests/etcd.yaml or one of the other manifests. 
3. Run `kubeadm init phase control-plane` or `kubeadm init phase etcd local`
4. Files have been recreated.

### PR DESCRIPTION
**Pull request by rcythr**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Implements the following change in behavior/feature:
(1) Prevents the recreation of control-plane and etcd manifests if the files already exist.

Fixes a these small issues:
(2) Moves the check for the etcd manifest to the etcd section so it only happens for local etcd.
(3) Fixes a typo in a docstring.

**Which issue(s) this PR fixes**: 
Fixes kubernetes/kubeadm#1679

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```
I put NONE for this being a user-facing change, but I'm not sure how strictly that's defined. I'll write a release note if we need one.

### CODE REVIEW

File: cmd/kubeadm/app/phases/controlplane/manifests.go [hunk 1]
```diff
@@ -124,11 +124,14 @@ func CreateStaticPodFiles(manifestDir string, cfg *kubeadmapi.ClusterConfigurati
 		}
 
 		// writes the StaticPodSpec to disk
-		if err := staticpodutil.WriteStaticPodToDisk(componentName, manifestDir, spec); err != nil {
-			return errors.Wrapf(err, "failed to create static pod manifest file for %q", componentName)
+		if !staticpodutil.CheckStaticPodExists(componentName, manifestDir) {
```
Comment (draveness) [inline cmd/kubeadm/app/phases/controlplane/manifests.go]: Please take a look at https://github.com/kubernetes/kubernetes/pull/80404#discussion_r305656699

**Review by draveness — COMMENTED** [review]
/ok-to-test

File: cmd/kubeadm/app/phases/controlplane/manifests.go (hunk 1 above)
Comment (rcythr) [reply cmd/kubeadm/app/phases/controlplane/manifests.go]: Agreed, a small refactoring of this to work like the check-or-create functions in the other phases makes sense.

**Review by fabriziopandini — COMMENTED** [review]
/hold
@rcythr if I got this right, this PR prevents kubeadm to override existing manifests.
What it is not clear to me, is how kubeadm is expected to execute upgrades after this PR is merged - when manifest already exists but kubeadm is expected to override them with a new release-. 

Could you kindly help me in understanding this bit?

**Comment by rcythr** [comment]
@fabriziopandini The upgrade process should be unaffected by this change because the upgrade code generates the manifest into a temporary directory instead of the manifests directory. It then moves the old manifest safely out of the way before moving the newly generated manifest into place. (https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/upgrade/staticpods.go#L194) 

All that said, I don't expect the upgrade will go smoothly for someone who requires customized changes like this (e.g. anyone patching manifests to support selinux). Hopefully kubernetes/kubeadm#1379 can fix that problem.

**Review by rosti — COMMENTED** [review]
Thank you @rcythr !

I am not very happy with this either. Sure, we have the upgrade case covered by the temp dirs, but that's almost a side effect. Plus, this is a significant behavior change and someone may be surprised by it.
Also, manifest overwrite can be avoided by skipping the control-plane and local etcd phases (which do nothing else, but setup the manifests) and the future is kubernetes/kubeadm#1379.

Although an opt-in manifest backup is always a good thing, users can actually do that by themselves (via plain old `cp`).

**Comment by rcythr** [comment]
Understood, I agree that https://github.com/kubernetes/kubeadm/issues/1379 is the best solution for this problem long term, and the short term workarounds are easy.

I'll close this PR.

**Pull request closed without merging**